packages/core/src/components/PlateControllerEffect.ts

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { focusAtom } from 'jotai-optics';
+import { useFocused } from 'slate-react';
+
+import {
+  plateControllerStore,
+  usePlateId,
+  usePlateSelectors,
+  usePlateStore,
+} from '../stores';
+import { usePlateControllerActions } from '../stores/plate-controller';
+
+export interface PlateControllerEffectProps {
+  id?: string;
+}
+
+export const PlateControllerEffect = ({
+  id: idProp,
+}: PlateControllerEffectProps) => {
+  const idFromStore = usePlateId();
+  const id = idProp ?? idFromStore;
+
+  const currentStoreAtom = React.useMemo(
+    () =>
+      focusAtom(plateControllerStore.atom.editorStores, (optic) =>
+        optic.prop(id)
+      ),
+    [id]
+  );
+  const setCurrentStore = usePlateControllerActions().atom(currentStoreAtom, {
+    warnIfNoStore: false,
+  });
+  const store = usePlateStore(id).store();
+
+  const primary = usePlateSelectors(id).primary();
+  const setPrimaryEdtorIds = usePlateControllerActions().primaryEditorIds({
```

Comment:
typo

Reply:
Nice catch! For some reason, "editor" is the word I typo most often in code, which is unfortunate given our line of work. 😆